mlflow/utils/environment.py

```diff
@@ -666,6 +667,30 @@ def _deduplicate_requirements(requirements):
     return [str(req) for req in deduped_reqs.values()]
 
 
+def _fetch_requirement_name(req: str) -> str:
+    try:
+        return Requirement(req).name
+    except InvalidRequirement:
+        return req
+
+
+def _remove_incompatible_requirements(requirements: list[str]) -> list[str]:
+    req_names = {_fetch_requirement_name(req) for req in requirements}
+    if "databricks-connect" in req_names and any(
+        x in req_names for x in ["pyspark", "pyspark-connect"]
+    ):
```

Comment:
```suggestion
    if "databricks-connect" in req_names and req_names.intersection({"pyspark", "pyspark-connect"}):
```

Can we use `intersection` here?